Dataset : CICIDS

Portion : Friday-WorkingHours-Afternoon-PortScan

File-name : Friday-WorkingHours-Afternoon-PortScan.pcap_ISCX.csv.zip



In [7]:
import pandas as pd

df = pd.read_csv("Friday-WorkingHours-Afternoon-PortScan.pcap_ISCX.csv")
print(df.shape)
df.head()

(286467, 79)


,Destination Port,Flow Duration,Total Fwd Packets,Total Backward Packets,Total Length of Fwd Packets,Total Length of Bwd Packets,Fwd Packet Length Max,Fwd Packet Length Min,Fwd Packet Length Mean,Fwd Packet Length Std,...,min_seg_size_forward,Active Mean,Active Std,Active Max,Active Min,Idle Mean,Idle Std,Idle Max,Idle Min,Label
0,22,1266342,41,44,2664,6954,456,0,64.975610,109.864573,...,32,0.0,0.0,0,0,0.0,0.0,0,0,BENIGN
1,22,1319353,41,44,2664,6954,456,0,64.975610,109.864573,...,32,0.0,0.0,0,0,0.0,0.0,0,0,BENIGN
2,22,160,1,1,0,0,0,0,0.000000,0.000000,...,32,0.0,0.0,0,0,0.0,0.0,0,0,BENIGN
3,22,1303488,41,42,2728,6634,456,0,66.536585,110.129945,...,32,0.0,0.0,0,0,0.0,0.0,0,0,BENIGN
4,35396,77,1,2,0,0,0,0,0.000000,0.000000,...,32,0.0,0.0,0,0,0.0,0.0,0,0,BENIGN


In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 286467 entries, 0 to 286466
Data columns (total 79 columns):
 #   Column                        Non-Null Count   Dtype  
---  ------                        --------------   -----  
 0    Destination Port             286467 non-null  int64  
 1    Flow Duration                286467 non-null  int64  
 2    Total Fwd Packets            286467 non-null  int64  
 3    Total Backward Packets       286467 non-null  int64  
 4   Total Length of Fwd Packets   286467 non-null  int64  
 5    Total Length of Bwd Packets  286467 non-null  int64  
 6    Fwd Packet Length Max        286467 non-null  int64  
 7    Fwd Packet Length Min        286467 non-null  int64  
 8    Fwd Packet Length Mean       286467 non-null  float64
 9    Fwd Packet Length Std        286467 non-null  float64
 10  Bwd Packet Length Max         286467 non-null  int64  
 11   Bwd Packet Length Min        286467 non-null  int64  
 12   Bwd Packet Length Mean       286467 non-nul

Data Status:

Rows (entries): 286467

Columns (features): 79

__

Integer columns: 50

Float columns: 28

Categorical (Label): 1

**ETL (Extract, Transform, Load)**

In [9]:
columns = [
    " Destination Port",
    " Flow Duration",
    " Total Fwd Packets",
    " Total Backward Packets",
    " Total Length of Fwd Packets",
    " Total Length of Bwd Packets",
    " Flow Bytes/s",
    " Flow Packets/s",
    " Packet Length Mean",
    " Packet Length Std",
    " SYN Flag Count",
    " ACK Flag Count",
    " Down/Up Ratio",
    " Average Packet Size",
    " Label"
]

In [11]:
df.columns = df.columns.str.strip()
cleaned_selection_columns = [col.strip() for col in columns]
df = df[cleaned_selection_columns]

In [12]:
df.to_csv("cleaned_network_data.csv", index=False)

Fact Table: network_flows

Field	Type

flow_duration    	measure

bytes_per_sec   	measure

packets_per_sec 	measure

avg_packet_size  	measure

packet_length_mean	measure

syn_flag_count	measure

is_attack	measure

